lib/ansible/modules/cloud/azure/azure_rm_aci.py

```diff
@@ -0,0 +1,503 @@
+                default=None
+            ),
+            containers=dict(
+                type='list',
+                required=True
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.tags = None
+        self.state = None
+        self.ip_address = None
+
+        self.containers = None
+
+        self.results = dict(changed=False, state=dict())
+
+        super(AzureRMContainerInstance, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                       supports_check_mode=True,
+                                                       supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        resource_group = None
+        response = None
+        results = dict()
+        to_be_updated = False
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {} not found'.format(self.resource_group))
+        if not self.location:
+            self.location = resource_group.location
+
+        response = self.get_aci()
+
+        if not response:
+            self.log("Container Group doesn't exist")
+
+            if self.state == 'absent':
+                self.log("Nothing to delete")
+            else:
+                to_be_updated = True
+        else:
+            self.log("Container instance already exists")
+
+            if self.state == 'absent':
+                self.delete_aci()
+                self.results['changed'] = True
+                self.log("ACI instance deleted")
+            elif self.state == 'present':
+                self.log("Need to check if container group has to be deleted or may be updated")
+                to_be_updated = self.check_need_update(response)
+                if to_be_updated:
+                    self.log('Deleting ACI instance before update')
+                    self.delete_aci()
+
+        if self.state == 'present':
+
+            self.log("Need to Create / Update the ACI instance")
+
+            if self.check_mode:
+                return self.results
+
+            if to_be_updated:
+                self.results['state'] = self.create_update_aci()
+                self.results['changed'] = True
+            else:
+                self.results['state'] = response
+
+            self.log("Creation / Update done")
+
+        return self.results
+
+    def create_update_aci(self):
+        '''
+        Creates or updates a container service with the specified configuration of orchestrator, masters, and agents.
+
+        :return: deserialized ACI instance state dictionary
+        '''
+        self.log("Creating / Updating the ACI instance {0}".format(self.name))
+
+        credentials = None
```

Comment:
`registryCredentials`? Or it is confused it is registry credential or azure credential